processing/src/main/java/org/apache/druid/segment/UnnestStorageAdapter.java

```diff
@@ -291,34 +306,26 @@ void add(@Nullable final Filter filter)
             }
           }
         }
-
-        if (requiredColumns.contains(outputColumnName)) {
-          // Try to move filter pre-correlate if possible.
-          final Filter newFilter = rewriteFilterOnUnnestColumnIfPossible(filter, inputColumn, inputColumnCapabilites);
-          if (newFilter != null) {
-            preFilters.add(newFilter);
-          } else {
-            postFilters.add(filter);
-          }
-        } else {
-          preFilters.add(filter);
-        }
+        preFilters.add(filter);
       }
     }
 
     final FilterSplitter filterSplitter = new FilterSplitter();
 
-    if (allowSet != null && !allowSet.isEmpty()) {
-      // Filter on input column if possible (it may be faster); otherwise use output column.
-      filterSplitter.add(new InDimFilter(inputColumn != null ? inputColumn : outputColumnName, allowSet));
-    }
-
     if (queryFilter instanceof AndFilter) {
       for (Filter filter : ((AndFilter) queryFilter).getFilters()) {
-        filterSplitter.add(filter);
+        filterSplitter.addPreFilter(filter);
+      }
+    } else {
+      filterSplitter.addPreFilter(queryFilter);
+    }
+
+    if (unnestFilter instanceof AndFilter) {
+      for (Filter filter : ((AndFilter) unnestFilter).getFilters()) {
+        filterSplitter.addPostFilterWithPreFilterIfRewritePossible(filter);
       }
     } else {
-      filterSplitter.add(queryFilter);
+      filterSplitter.addPostFilterWithPreFilterIfRewritePossible(unnestFilter);
     }
```

Comment:
We break it up in the case of AND so that we can find out if one of them can be added to the pre filters through re-write. Adding to pre-filters takes place inside `addPostFilterWithPreFilterIfRewritePossible` method

Reply:
This can be made without breaking up, refactored